app/components/Category.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useContext, useEffect, useRef, useState } from "react";
+import { PLATFORMS } from "../config/platforms";
+import { PlatformGroupSpec, STAMP_PROVIDERS } from "../config/providers";
+import { LoadingCard } from "./LoadingCard";
+import { GenericPlatform } from "./GenericPlatform";
+import { PLATFORM_ID, PROVIDER_ID } from "@gitcoin/passport-types";
+import { CeramicContext } from "../context/ceramicContext";
+import { PlatformCard } from "./PlatformCard";
+import { PlatformScoreSpec, ScorerContext } from "../context/scorerContext";
+import { Disclosure } from "@headlessui/react";
+import { DropDownIcon } from "./DropDownIcon";
+import { SideBarContent } from "./SideBarContent";
+import { Drawer, DrawerOverlay, useDisclosure } from "@chakra-ui/react";
+
+export type Category = {
+  name: string;
+  description: string;
+  sortedPlatforms: PlatformScoreSpec[];
+};
+
+export type CategoryProps = {
+  isLoading?: boolean;
+  className?: string;
+  initialOpen?: boolean;
+  category: Category;
+};
+
+const cardClassName = "col-span-2 md:col-span-1 lg:col-span-1 xl:col-span-1";
+
+type SelectedProviders = Record<PLATFORM_ID, PROVIDER_ID[]>;
+
+export const getStampProviderIds = (platform: PLATFORM_ID): PROVIDER_ID[] => {
+  return (
+    STAMP_PROVIDERS[platform]?.reduce((all, stamp) => {
+      return all.concat(stamp.providers?.map((provider) => provider.name as PROVIDER_ID));
+    }, [] as PROVIDER_ID[]) || []
+  );
+};
+
+export const Category = ({
+  className,
+  category,
+  isLoading = false,
+  initialOpen = true,
+}: CategoryProps): JSX.Element => {
+  const { allProvidersState, allPlatforms } = useContext(CeramicContext);
+  const [dropDownOpen, setDropDownOpen] = useState<boolean>(false);
+  const openRef = React.useRef(dropDownOpen);
+  openRef.current = dropDownOpen;
+
+  const [panelMounted, setPanelMounted] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (initialOpen) {
+      handleOpen();
+    }
+  }, [initialOpen]);
+
+  const handleOpen = () => {
+    setPanelMounted(true);
+  };
+
+  useEffect(() => {
+    setDropDownOpen(panelMounted);
+  }, [panelMounted]);
+
+  const handleClose = () => {
+    setDropDownOpen(false);
+    setTimeout(() => {
+      const isOpen = openRef.current;
+      if (!isOpen) setPanelMounted(false);
+    }, 150);
+  };
+
+  const handleClick = () => {
+    if (dropDownOpen) handleClose();
+    else handleOpen();
+  };
+
```

Comment:
Using an `useCallback` ( see https://react.dev/reference/react/useCallback ) on any function defined in the component probably makes sense